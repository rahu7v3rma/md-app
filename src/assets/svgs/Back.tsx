import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Path, Svg } from 'react-native-svg';

type Props = {
    width?: number;
    height?: number;
};

const BackIcon: FunctionComponent<Props> = ({
    width = 8,
    height = 14
}: Props) => {
    return (
        <Svg
            width={width}
            height={height}
            style={style.root}
            viewBox="0 0 8 14"
            fill="none"
        >
            <Path
                d="M5.97999 0.319993L2.76999 3.52999L0.79999 5.48999C-0.0300097 6.31999 -0.0300097 7.66999 0.79999 8.49999L5.97999 13.68C6.65999 14.36 7.81999 13.87 7.81999 12.92V7.30999V1.07999C7.81999 0.119992 6.65999 -0.360007 5.97999 0.319993Z"
                fill="#271A51"
            />
        </Svg>
    );
};

const style = StyleSheet.create({
    root: {
        width: 'auto',
        height: 'auto'
    }
});

export default BackIcon;
