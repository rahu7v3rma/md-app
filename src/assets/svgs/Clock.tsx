import React, { FunctionComponent } from 'react';
import { Path, Svg } from 'react-native-svg';

type Props = {
    width?: number;
    height?: number;
    color?: string;
};

const ClockIcon: FunctionComponent<Props> = ({
    width = 11,
    height = 12,
    color = '#6E51D0'
}: Props) => {
    return (
        <Svg width={width} height={height} viewBox="0 0 11 12" fill="none">
            <Path
                d="M0.776001 6.76999C0.776001 4.16599 2.896 2.04599 5.5 2.04599C8.104 2.04599 10.224 4.171 10.224 6.776C10.224 9.38 8.104 11.5 5.5 11.5C2.896 11.5 0.776001 9.37499 0.776001 6.76999ZM5.5 7.354C5.923 7.354 6.271 7.00601 6.271 6.58301V3.66699C6.271 3.24299 5.923 2.896 5.5 2.896C5.077 2.896 4.729 3.24299 4.729 3.66699V6.58301C4.729 7.00601 5.077 7.354 5.5 7.354Z"
                fill={color}
                stroke={color}
                stroke-width="0.666667"
            />
            <Path
                d="M7.18597 0.678986H3.81403C3.76503 0.678986 3.72699 0.64201 3.72699 0.59201C3.72699 0.53701 3.77103 0.5 3.81403 0.5H7.18597C7.23497 0.5 7.27197 0.537006 7.27197 0.587006C7.27197 0.642006 7.22897 0.678986 7.18597 0.678986Z"
                fill={color}
                stroke={color}
                stroke-width="0.666667"
            />
        </Svg>
    );
};

export default ClockIcon;
