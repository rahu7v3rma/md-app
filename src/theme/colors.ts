// theme colors
const PRIMARY = '#72996B';
const PRIMARY_LIGHT = '#D3E29F40';
const BACKGROUND = '#F5F5F5';
const SHEET_BACKGROUND_COLOR = 'rgba(0,0,0,0.5)';
const BACKGROUND_LIGHTEST = '#FAFAFF';
const LOG_PAGE_BACKGROUND_COLOR = '#f5f4ee';

// INPUT
const INPUT = '#D3E6F8';
const INPUT_BG = '#EEF4FA';
const INPUT_ERROR_BG = '#FFECEC';
const INPUT_ERROR_BORDER = '#F9D4D4';
const INPUT_ERROR_PLACEHOLDER = '#D05151';
const INPUT_BORDER = '#D3E29F';

// PROGRESS BAR
const PROGRESS_BAR_FILL = '#4DBA0B';
const PROGRESS_BAR_FILL_BORDER = '#3A9A00';
const PROGRESS_BAR_ACTIVITY_INDICATOR = PRIMARY;
const PROGRESS_BAR_FILL_GREEN = '#72996B';
const PROGRESS_BAR_FILL_DISABLE = '#BFC4C9';
const PROGRESS_BAR_TRACK_GREEN = '#EDF0DA';

// PROGRESS CIRCLE
const PROGRESS_CIRCLE_COLOR = '#f00';
const PROGRESS_CIRCLE_SHADOW_COLOR = '#999';
const PROGRESS_CIRCLE_BG_COLOR = '#e9e9ef';
const PROGRESS_CIRCLE_START_BALL_COLOR = '#72996B';
const PROGRESS_CIRCLE_START_BALL_DISABLE_COLOR = '#BDBECA';
const PROGRESS_CIRCLE_INNER_BALL_COLOR = '#5B37BE';
const PROGRESS_CIRCLE_INNER_BALL_DISABLE_COLOR = '#A4AAAF';

// SLIDER
const SLIDER_BG = 'rgba(120, 120, 128, 0.32)';

// BUTTON
const BUTTON_BORDER = '#D3E6F8';
const GREEN_CHECK_DARK = '#D3E6F8';
const BUTTON_LIGHTER_BORDER = '#f4f8e7';
const BUTTON_LIGHTER_BACKGROUND = '#f4f8e7';
const BUTTON_PRIMARY_BORDER = '#5A3DBF';
const BUTTON_DISABLED_BG = '#cccccc';
const BUTTON_DISABLED_BORDER = '#D6D6D6';
const BUTTON_IMAGE_BACKGROUND = '#F0E957';
const BUTTON_ICON_BACKGROUND = '#D3E29F40';
const BUTTON_RED_BACKGROUND = '#B16663';
const BUTTON_SUCCESS_BACKGROUND = '#4DBA0B';
const BUTTON_GREEN_BACKGROUND = '#72996B';
const BUTTON_WHITE_BACKGROUND = '#FFFFFF';
const BUTTON_DARK_RED_BACKGROUND = '#CD4049';

const TEXT_BLACK = '#271A51'; // more of a dark-blue
const TEXT_DARK_BLACK = '#1F1E22';
const TEXT_BLACK_GRAY = '#4D4F4D';
const TEXT_GRAY_BLACK = '#4D4F4D';
const TEXT_WHITE = '#FFFFFF';
const TEXT_GREEN = '#72996B';
const TEXT_CHARCOAL = '#4D4F4D';
const TEXT_DUSKY_OLIVE = '#847F60';
const TEXT_ERROR = '#D05151';
const TEXT_LIGHTER = 'rgba(235, 235, 245, 0.6)';
const TEXT_DISABLED = '#D6D6D6';
const TEXT_GRAY = '#A4AAAF';
const TEXT_GRAY_BASE = '#58595A';
const TEXT_PRIMARY = PRIMARY;
const TEXT_PURPLE = '#6E51D0';
const TEXT_LINK = '#3ED1FF';
const TEXT_DARK_GREEN = '#72996B';
const TEXT_DARKER = '#4D4F4D';
const TEXT_MAIN_DARKER = '#4D4F4D';

//ICONS
const EYE_ICON = '#6E51D0';
const EYE_ICON_GREEN = '#72996B';
const EYE_ICON_ERROR_BORDER = '#D05151';
const GREEN_CHECK_MARK = '#4DBA0B';

// toast messages
const ERROR_BACKGROUND = '#D05151';
const SUCCESS_BACKGROUND = '#4DBA0B';
const INFO_BACKGROUND = '#A4AAAF';

//Modals
const MODAL_BACKDROP = 'rgba(0,0,0,0.5)';
const MODAL_CLOSE_HANDLE = '#F5F8FB';
const MODAL_ITEM_DIVIDER = '#E6E8EC';

//EXTRAS
const SUCCESS_LIGHTEST = '#ECF8E6';
const SUCCESS_LIGHTER = '#DFEED8';
const DARK_CHARCOAL = '#1F1E22';
const WHITE = '#FFFFFF';
const BLACK = '#000000';
const BLACK_LIGHTER = 'rgba(37, 37, 37, 0.6)';
const JOURNEY_BACKGROUND = '#CEE1FE';
const JOURNEY_SECONDARY = '#ADCAF4';
const BLOCK_IN_PROGRESS = PRIMARY;
//const BLOCK_COMPLETE = '#4DBA0B';
const ERROR_LIGHTEST = '#FFECEC';
const ERROR_LIGHTER = '#F9D4D4';
const ERROR_BASE = '#D05151';
const BLACK_SHADOW = '#171717';
const CARD_COMPLETED = '#eff8e7';
const CARD_INPROGRESS = '#f0f4f8';
const PAGE_BG = '#F5F8FB';
const LIGHT_BACK_COLOR = '#D3E29F40';
const BLACK_CARDS = 'rgba(0, 0, 0, 0.3)';
const HEADER_SHADOW = 'rgba(0, 0, 0, 0.03)';
const GRAY_SHADOW = '#0000000D';
const BORDER_COLOR = '#E1E6C2';
const TRANSPARENT_COLOR = 'transparent';
const BADGE_RED = '#F50157';

//SWITCH
const SWITCH_TOGGLED = '#72996B';
const SWITCH_BUTTON = WHITE;

// pagination
const ACTIVE_PAGINATION_DOT = PRIMARY;
const INACTIVE_PAGINATION_DOT = PRIMARY_LIGHT;

export const Colors = {
    theme: {
        primary: PRIMARY,
        primary_light: PRIMARY_LIGHT,
        app_background: BACKGROUND,
        app_sheet_background_color: SHEET_BACKGROUND_COLOR,
        app_background_lightest: BACKGROUND_LIGHTEST,
        log_page_background_color: LOG_PAGE_BACKGROUND_COLOR
    },
    text: {
        black: TEXT_BLACK,
        black_gray: TEXT_BLACK_GRAY,
        dark_green: TEXT_DARK_GREEN,
        text_gray_black: TEXT_GRAY_BLACK,
        dark_black: TEXT_DARK_BLACK,
        white: TEXT_WHITE,
        error: TEXT_ERROR,
        green: TEXT_GREEN,
        charcoal: TEXT_CHARCOAL,
        dusky_olive: TEXT_DUSKY_OLIVE,
        text_lighter: TEXT_LIGHTER,
        text_disabled: TEXT_DISABLED,
        gray: TEXT_GRAY,
        gray_Base: TEXT_GRAY_BASE,
        primary: TEXT_PRIMARY,
        purple: TEXT_PURPLE,
        link: TEXT_LINK,
        darker: TEXT_DARKER,
        mainDarker: TEXT_MAIN_DARKER
    },
    input: {
        error_bg: INPUT_ERROR_BG,
        error_border: INPUT_ERROR_BORDER,
        app_input: INPUT,
        app_input_bg: INPUT_BG,
        input_placeholder_error: INPUT_ERROR_PLACEHOLDER,
        input_border: INPUT_BORDER,
        light_back_color: LIGHT_BACK_COLOR
    },
    button: {
        app_button_border: BUTTON_BORDER,
        app_button_lighter_background: BUTTON_LIGHTER_BACKGROUND,
        app_button_lighter_border: BUTTON_LIGHTER_BORDER,
        app_button_primary_border: BUTTON_PRIMARY_BORDER,
        app_button_disabled_bg: BUTTON_DISABLED_BG,
        app_button_disabled_border: BUTTON_DISABLED_BORDER,
        app_button_background: BUTTON_IMAGE_BACKGROUND,
        app_icon_button_background: BUTTON_ICON_BACKGROUND,
        app_button_red_background: BUTTON_RED_BACKGROUND,
        app_button_success_background: BUTTON_SUCCESS_BACKGROUND,
        app_button_green_background: BUTTON_GREEN_BACKGROUND,
        app_button_white_background: BUTTON_WHITE_BACKGROUND,
        app_btn_light_color: LIGHT_BACK_COLOR,
        app_button_dark_red_background: BUTTON_DARK_RED_BACKGROUND
    },
    icons: {
        eye: EYE_ICON,
        eyeGreen: EYE_ICON_GREEN,
        eye_error_border: EYE_ICON_ERROR_BORDER,
        green_check_mark: GREEN_CHECK_MARK,
        green_dark_check: GREEN_CHECK_DARK
    },
    modals: {
        backdrop: MODAL_BACKDROP,
        close_handle: MODAL_CLOSE_HANDLE,
        divider: MODAL_ITEM_DIVIDER
    },
    toast: {
        errorBackground: ERROR_BACKGROUND,
        successBackground: SUCCESS_BACKGROUND,
        infoBackground: INFO_BACKGROUND
    },
    progress: {
        fillWhite: TEXT_WHITE,
        fill: PROGRESS_BAR_FILL,
        fill_border: PROGRESS_BAR_FILL_BORDER,
        activity_indicator: PROGRESS_BAR_ACTIVITY_INDICATOR,
        fillGreen: PROGRESS_BAR_FILL_GREEN,
        fillDisable: PROGRESS_BAR_FILL_DISABLE,
        trackColor: PROGRESS_BAR_TRACK_GREEN,
        headerProgressColor: PROGRESS_CIRCLE_INNER_BALL_COLOR
    },
    slider: {
        bg: SLIDER_BG
    },
    extras: {
        success_lightest: SUCCESS_LIGHTEST,
        success_lighter: SUCCESS_LIGHTER,
        white: WHITE,
        black: BLACK,
        black_lighter: BLACK_LIGHTER,
        blue_background: JOURNEY_BACKGROUND,
        blue_secondary: JOURNEY_SECONDARY,
        block_in_progress: BLOCK_IN_PROGRESS,
        block_complete: BLOCK_IN_PROGRESS,
        error_lighter: ERROR_LIGHTER,
        error_lightest: ERROR_LIGHTEST,
        error_base: ERROR_BASE,
        black_shadow: BLACK_SHADOW,
        card_completed: CARD_COMPLETED,
        card_inProgress: CARD_INPROGRESS,
        page_bg: PAGE_BG,
        blackCards: BLACK_CARDS,
        headerShadow: HEADER_SHADOW,
        dark_charcoal: DARK_CHARCOAL,
        block_locked: INPUT_BORDER,
        gray_shadow: GRAY_SHADOW,
        border_color: BORDER_COLOR,
        transparent: TRANSPARENT_COLOR,
        badgeRed: BADGE_RED
    },
    pagination: {
        active_dot: ACTIVE_PAGINATION_DOT,
        inactive_dot: INACTIVE_PAGINATION_DOT
    },
    progressCircle: {
        color: PROGRESS_CIRCLE_COLOR,
        shadow_color: PROGRESS_CIRCLE_SHADOW_COLOR,
        bg_color: PROGRESS_CIRCLE_BG_COLOR,
        start_ball_color: PROGRESS_CIRCLE_START_BALL_COLOR,
        start_ball_disable_color: PROGRESS_CIRCLE_START_BALL_DISABLE_COLOR,
        inner_ball_color: PRIMARY,
        inner_ball_disable_color: PROGRESS_CIRCLE_INNER_BALL_DISABLE_COLOR
    },
    switch: {
        button: SWITCH_BUTTON,
        toggled: SWITCH_TOGGLED
    },
    statusBar: {
        white: WHITE
    }
};
