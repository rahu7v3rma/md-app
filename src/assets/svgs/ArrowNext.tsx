import React, { FunctionComponent } from 'react';
import { Path, Svg } from 'react-native-svg';

import { Colors } from '@/theme/colors';

type Props = {
    width?: number;
    height?: number;
    color?: string;
};

const ArrowNext: FunctionComponent<Props> = ({
    width = 24,
    height = 24,
    color = Colors.text.gray
}: Props) => {
    return (
        <Svg width={width} height={height} viewBox="0 0 24 24" fill="none">
            <Path
                d="M9.5 7L14.5 12L9.5 17"
                stroke={color}
                strokeWidth={2}
                strokeLinecap="round"
                strokeLinejoin="round"
            />
        </Svg>
    );
};

export default ArrowNext;
