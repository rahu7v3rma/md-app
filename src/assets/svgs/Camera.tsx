import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Mask, Path, Svg } from 'react-native-svg';

import { Colors } from '@/theme/colors';

type Props = {
    width?: number;
    height?: number;
};

const Camera: FunctionComponent<Props> = ({
    width = 20,
    height = 20
}: Props) => {
    return (
        <Svg
            width={width}
            height={height}
            style={style.root}
            viewBox="0 0 20 20"
            fill="none"
        >
            <Mask id="path-1-inside-1_1_564" fill={'white'} />
            <Path d="M11.7667 3.33332L13.2917 4.99999H16.6667V15H3.33333V4.99999H6.70833L8.23333 3.33332H11.7667ZM12.5 1.66666H7.5L5.975 3.33332H3.33333C2.41666 3.33332 1.66666 4.08332 1.66666 4.99999V15C1.66666 15.9167 2.41666 16.6667 3.33333 16.6667H16.6667C17.5833 16.6667 18.3333 15.9167 18.3333 15V4.99999C18.3333 4.08332 17.5833 3.33332 16.6667 3.33332H14.025L12.5 1.66666ZM10 7.49999C11.375 7.49999 12.5 8.62499 12.5 9.99999C12.5 11.375 11.375 12.5 10 12.5C8.625 12.5 7.5 11.375 7.5 9.99999C7.5 8.62499 8.625 7.49999 10 7.49999ZM10 5.83332C7.7 5.83332 5.83333 7.69999 5.83333 9.99999C5.83333 12.3 7.7 14.1667 10 14.1667C12.3 14.1667 14.1667 12.3 14.1667 9.99999C14.1667 7.69999 12.3 5.83332 10 5.83332Z" />

            <Path
                d="M11.7667 3.33332L13.2917 4.99999H16.6667V15H3.33333V4.99999H6.70833L8.23333 3.33332H11.7667ZM12.5 1.66666H7.5L5.975 3.33332H3.33333C2.41666 3.33332 1.66666 4.08332 1.66666 4.99999V15C1.66666 15.9167 2.41666 16.6667 3.33333 16.6667H16.6667C17.5833 16.6667 18.3333 15.9167 18.3333 15V4.99999C18.3333 4.08332 17.5833 3.33332 16.6667 3.33332H14.025L12.5 1.66666ZM10 7.49999C11.375 7.49999 12.5 8.62499 12.5 9.99999C12.5 11.375 11.375 12.5 10 12.5C8.625 12.5 7.5 11.375 7.5 9.99999C7.5 8.62499 8.625 7.49999 10 7.49999ZM10 5.83332C7.7 5.83332 5.83333 7.69999 5.83333 9.99999C5.83333 12.3 7.7 14.1667 10 14.1667C12.3 14.1667 14.1667 12.3 14.1667 9.99999C14.1667 7.69999 12.3 5.83332 10 5.83332Z"
                fill={Colors.theme.primary}
            />
            <Path
                d="M13.2917 4.99999L13.1441 5.135L13.2036 5.19999H13.2917V4.99999ZM16.6667 4.99999H16.8667V4.79999H16.6667V4.99999ZM16.6667 15V15.2H16.8667V15H16.6667ZM3.33333 15H3.13333V15.2H3.33333V15ZM3.33333 4.99999V4.79999H3.13333V4.99999H3.33333ZM6.70833 4.99999V5.19999H6.79642L6.85588 5.135L6.70833 4.99999ZM8.23333 3.33332V3.13332H8.14524L8.08578 3.19831L8.23333 3.33332ZM12.5 1.66666L12.6476 1.53165L12.5881 1.46666H12.5V1.66666ZM7.5 1.66666V1.46666H7.41191L7.35244 1.53165L7.5 1.66666ZM5.975 3.33332V3.53332H6.06309L6.12255 3.46833L5.975 3.33332ZM14.025 3.33332L13.8774 3.46833L13.9369 3.53332H14.025V3.33332ZM11.6191 3.46833L13.1441 5.135L13.4392 4.86498L11.9142 3.19831L11.6191 3.46833ZM13.2917 5.19999H16.6667V4.79999H13.2917V5.19999ZM16.4667 4.99999V15H16.8667V4.99999H16.4667ZM16.6667 14.8H3.33333V15.2H16.6667V14.8ZM3.53333 15V4.99999H3.13333V15H3.53333ZM3.33333 5.19999H6.70833V4.79999H3.33333V5.19999ZM6.85588 5.135L8.38088 3.46833L8.08578 3.19831L6.56078 4.86498L6.85588 5.135ZM8.23333 3.53332H11.7667V3.13332H8.23333V3.53332ZM12.5 1.46666H7.5V1.86666H12.5V1.46666ZM7.35244 1.53165L5.82744 3.19831L6.12255 3.46833L7.64755 1.80167L7.35244 1.53165ZM5.975 3.13332H3.33333V3.53332H5.975V3.13332ZM3.33333 3.13332C2.30621 3.13332 1.46666 3.97287 1.46666 4.99999H1.86666C1.86666 4.19378 2.52712 3.53332 3.33333 3.53332V3.13332ZM1.46666 4.99999V15H1.86666V4.99999H1.46666ZM1.46666 15C1.46666 16.0271 2.30621 16.8667 3.33333 16.8667V16.4667C2.52712 16.4667 1.86666 15.8062 1.86666 15H1.46666ZM3.33333 16.8667H16.6667V16.4667H3.33333V16.8667ZM16.6667 16.8667C17.6938 16.8667 18.5333 16.0271 18.5333 15H18.1333C18.1333 15.8062 17.4729 16.4667 16.6667 16.4667V16.8667ZM18.5333 15V4.99999H18.1333V15H18.5333ZM18.5333 4.99999C18.5333 3.97287 17.6938 3.13332 16.6667 3.13332V3.53332C17.4729 3.53332 18.1333 4.19378 18.1333 4.99999H18.5333ZM16.6667 3.13332H14.025V3.53332H16.6667V3.13332ZM14.1726 3.19831L12.6476 1.53165L12.3524 1.80167L13.8774 3.46833L14.1726 3.19831ZM10 7.69999C11.2645 7.69999 12.3 8.73545 12.3 9.99999H12.7C12.7 8.51453 11.4855 7.29999 10 7.29999V7.69999ZM12.3 9.99999C12.3 11.2645 11.2645 12.3 10 12.3V12.7C11.4855 12.7 12.7 11.4854 12.7 9.99999H12.3ZM10 12.3C8.73545 12.3 7.7 11.2645 7.7 9.99999H7.3C7.3 11.4854 8.51454 12.7 10 12.7V12.3ZM7.7 9.99999C7.7 8.73545 8.73545 7.69999 10 7.69999V7.29999C8.51454 7.29999 7.3 8.51453 7.3 9.99999H7.7ZM10 5.63332C7.58954 5.63332 5.63333 7.58953 5.63333 9.99999H6.03333C6.03333 7.81045 7.81045 6.03332 10 6.03332V5.63332ZM5.63333 9.99999C5.63333 12.4104 7.58954 14.3667 10 14.3667V13.9667C7.81045 13.9667 6.03333 12.1895 6.03333 9.99999H5.63333ZM10 14.3667C12.4105 14.3667 14.3667 12.4104 14.3667 9.99999H13.9667C13.9667 12.1895 12.1895 13.9667 10 13.9667V14.3667ZM14.3667 9.99999C14.3667 7.58953 12.4105 5.63332 10 5.63332V6.03332C12.1895 6.03332 13.9667 7.81045 13.9667 9.99999H14.3667Z"
                fill={Colors.theme.primary}
                mask="url(#path-1-inside-1_1_564)"
            />
        </Svg>
    );
};

const style = StyleSheet.create({
    root: {
        width: 'auto',
        height: 'auto'
    }
});

export default Camera;
