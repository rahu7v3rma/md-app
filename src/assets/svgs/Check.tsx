import React, { FunctionComponent } from 'react';
import { Path, Rect, Svg } from 'react-native-svg';

import { Colors } from '@/theme/colors';

type Props = {
    width?: number;
    height?: number;
    color?: string;
};

const CheckIcon: FunctionComponent<Props> = ({
    width = 97,
    height = 96,
    color = Colors.extras.white
}: Props) => {
    return (
        <Svg width={width} height={height} viewBox="0 0 97 96" fill="none">
            <Rect
                x={0.5}
                width={width}
                height={height}
                rx={32}
                fill={Colors.theme.primary}
            />
            <Path
                d="M62.5 38l-20 20-8-8"
                stroke={color}
                strokeWidth={4}
                strokeLinecap="round"
                strokeLinejoin="round"
            />
        </Svg>
    );
};

export default CheckIcon;
