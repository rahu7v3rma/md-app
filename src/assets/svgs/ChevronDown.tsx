import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import { Path, Svg } from 'react-native-svg';

import { Colors } from '@/theme/colors';

type Props = {
    width?: number;
    height?: number;
};

const ChevronDown: FunctionComponent<Props> = ({
    width = 12,
    height = 8
}: Props) => {
    return (
        <Svg
            width={width}
            height={height}
            viewBox="0 0 12 8"
            fill="none"
            style={style.root}
        >
            <Path
                d="M11 1.5L6 6.5L1 1.5"
                stroke={Colors.theme.primary}
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
            />
        </Svg>
    );
};

const style = StyleSheet.create({
    root: {
        width: 'auto',
        height: 'auto'
    }
});

export default ChevronDown;
