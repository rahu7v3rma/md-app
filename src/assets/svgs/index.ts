import ArrowNext from './ArrowNext';
import BackIcon from './Back';
import CheckIcon from './Check';
import ChevronDown from './ChevronDown';
import ChoosePhoto from './ChoosePhoto';
import ClockIcon from './Clock';
import CrossIcon from './Cross';
import EditProfile from './EditProfile';
import EmailIcon from './EmailIcon';
import EyeIcon from './Eye';
import EyeOpenIcon from './EyeOpen';
import Heart from './Heart';
import HighFive from './HighFive';
import Like from './Like';
import LoginLogoIcon from './LoginLogo';
import LogoutIcon from './logout';
import Muscle from './Muscle';
import NoData from './NoData';
import BellIcon from './notification';
import PlayIcon from './Play';
import Plus from './Plus';
import ProfileIcon from './ProfileIcon';
import SadFace from './SadFace';
import SearchIcon from './Search';
import Setting from './Setting';
import SliderMarker from './SliderMarker';
import SmileyFace from './SmileyFace';
import TakePhoto from './TakePhoto';
import WolfIcon from './Wolf';
import WolfIconNotes from './WolficonNotes';
import WolfIconDance from './WolfiocnDance';

export {
    BackIcon,
    BellIcon,
    CheckIcon,
    ChevronDown,
    CrossIcon,
    EditProfile,
    EmailIcon,
    EyeIcon,
    EyeOpenIcon,
    Heart,
    HighFive,
    Like,
    LoginLogoIcon,
    LogoutIcon,
    Muscle,
    PlayIcon,
    ProfileIcon,
    SadFace,
    SliderMarker,
    SmileyFace,
    WolfIcon,
    Plus,
    NoData,
    ClockIcon,
    SearchIcon,
    WolfIconNotes,
    WolfIconDance,
    ArrowNext,
    ChoosePhoto,
    TakePhoto,
    Setting
};
