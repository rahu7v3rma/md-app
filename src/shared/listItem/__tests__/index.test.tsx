import React from 'react';
import { StyleSheet } from 'react-native';
import renderer, { act } from 'react-test-renderer';

import { BellIcon } from '@/assets/svgs';
import ListItem from '@/shared/listItem';
import { Colors } from '@/theme/colors';

const listItemStyles = StyleSheet.create({
    container: {
        height: 56,
        backgroundColor: Colors.extras.white,
        flexDirection: 'row',
        alignItems: 'center',
        borderRadius: 16,
        marginBottom: 16
    },
    icon: {
        width: '15%',
        alignItems: 'center'
    },
    title: {
        width: '75%'
    },
    arrow: {
        width: '10%'
    }
});

it('listItem Snapshot Testing with Jest', async () => {
    const tree = renderer.create(<ListItem />);
    expect(tree).toMatchSnapshot();
});

it('listItem renders without crashing', async () => {
    const tree = renderer.create(<ListItem />);
    expect(tree.root).toBeTruthy();
});

it('listItem renders the provided title correctly', async () => {
    const title = 'Test Title';
    const tree = renderer.create(<ListItem title={title} />);
    const listItemTitle = tree.root.findByProps({
        testID: 'listItemTitle'
    }).props;
    expect(listItemTitle.children).toBe(title);
});

it('listItem renders the provided Icon component correctly', async () => {
    const tree = renderer.create(<ListItem Icon={BellIcon} />);
    const listItemIcon = tree.root.findByProps({
        testID: 'listItemIcon'
    }).props;
    expect(listItemIcon.children).toMatchSnapshot();
});

it('listItem calls onPress when the component is pressed', async () => {
    const onListItemPress = jest.fn();
    const tree = renderer.create(<ListItem onPress={onListItemPress} />);

    const listItemButton = tree.root.findByProps({
        testID: 'listItemButton'
    }).props;
    await act(() => listItemButton.onPress());
    expect(onListItemPress).toHaveBeenCalledTimes(1);
});

it('listItem applies the correct styles', async () => {
    const tree = renderer.create(<ListItem />);
    const listItemButton = tree.root.findByProps({
        testID: 'listItemButton'
    }).props;
    expect(listItemButton.style).toStrictEqual(listItemStyles.container);

    const listItemIcon = tree.root.findByProps({
        testID: 'listItemIcon'
    }).props;
    expect(listItemIcon.style).toStrictEqual(listItemStyles.icon);

    const listItemTitleContainer = tree.root.findByProps({
        testID: 'listItemTitleContainer'
    }).props;
    expect(listItemTitleContainer.style).toStrictEqual(listItemStyles.title);

    const listItemArrow = tree.root.findByProps({
        testID: 'listItemArrow'
    }).props;
    expect(listItemArrow.style).toStrictEqual(listItemStyles.arrow);
});
