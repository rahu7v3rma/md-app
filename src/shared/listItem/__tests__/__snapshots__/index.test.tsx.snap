// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listItem Snapshot Testing with Jest 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 16,
      "flexDirection": "row",
      "height": 56,
      "marginBottom": 16,
      "opacity": 1,
    }
  }
  testID="listItemButton"
>
  <View
    style={
      {
        "alignItems": "center",
        "width": "15%",
      }
    }
    testID="listItemIcon"
  />
  <View
    style={
      {
        "width": "75%",
      }
    }
    testID="listItemTitleContainer"
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          undefined,
          {
            "color": "#271A51",
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 22,
          },
        ]
      }
      testID="listItemTitle"
    />
  </View>
  <View
    style={
      {
        "width": "10%",
      }
    }
    testID="listItemArrow"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={12}
      bbWidth={8}
      fill="none"
      focusable={false}
      height={12}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 12,
            "width": 8,
          },
        ]
      }
      vbHeight={12}
      vbWidth={8}
      width={8}
      xml="<svg width="8" height="12" viewBox="0 0 8 12" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M1.5 0.999999L6.5 6L1.5 11" stroke="#6E51D0" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M1.5 0.999999L6.5 6L1.5 11"
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4285419984}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`listItem renders the provided Icon component correctly 1`] = `<BellIcon />`;
